govcd/adminvdc.go

```diff
@@ -23,6 +25,174 @@ func NewAdminVdc(cli *Client) *AdminVdc {
 	}
 }
 
+// vdcVersionedFuncs holds interface of functions
+type vdcVersionedFuncs struct {
```

Comment:
`vdcVersionedFuncs` describes what the type is at a grammatical level, rather than at functional level.
The reject3ed term `vdcProducer` was showing the purpose of the structure, while the current name only shows that it contains functions.

Reply:
and what I can do about it?

Reply:
I still think that a purposeful name such as `provider` conveys the meaning better than `Funcs`.  How about changing it to `vdcCRUD` ? At least it suggests some purpose.
cc: @lvirbalas, @Didainius

Reply:
There was a long discussion about this here:
https://github.com/vmware/go-vcloud-director/pull/285#discussion_r376947094

There `vdcCrud` was proposed too, but at the end we agreed on `vdcVersionedFuncs`, because:
- It has a word "versioned", which makes it instantly clear what the pattern's purpose is
- With such name not only CRUD functions could land here if needed